docs/NCN Lifecycle Hooks.md

```diff
@@ -0,0 +1,19 @@
+#### NCN Lifecycle Hooks
```

Comment:
Should you add underscore to title and remove spaces for easier file referance?

Reply:
it is the 3rd party lib we used that requires file name matches title ... :(